src/service.js

```diff
@@ -126,9 +126,9 @@ export function getServicePromiseOrNull(win, id) {
  * expected to return the service.
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrDoc
  * @param {string} id of the service.
- * @param {function(!AmpDoc):!Object=} opt_factory Should create the service
- *     if it does not exist yet. If the factory is not given, it is an error
- *     if the service does not exist yet.
+ * @param {function(!./service/ampdoc-impl.AmpDoc):!Object=} opt_factory
```

Comment:
Can we do here:
```
@param {function(!AmpDoc):T=} opt_factory
@return {T}
@template T
```
?

That's what we do for `getService`, so this should be similar.